src/services/identity/SitesService.ts

```diff
@@ -78,6 +84,107 @@ class SitesService {
     return user.site_members.map((site) => site.repo?.name)
   }
 
+  async getCommitAuthorEmail(commit: GitHubCommitData) {
+    const { message } = commit
+
+    // Commit message created as part of phase 2 identity
+    if (message.startsWith("{") && message.endsWith("}")) {
+      const { userId }: { userId: string } = JSON.parse(message)
+      const user = await this.usersService.findById(userId)
+
+      if (user) {
+        return user.email
+      }
+    }
+
+    // Legacy style of commits, or if the user is not found
+    const {
+      author: { email: authorEmail },
+    } = commit
+    return authorEmail
+  }
+
+  async getUrlsOfSite(
+    sessionData: UserWithSiteSessionData,
+    urlTypes: SiteUrlTypes[]
+  ): Promise<SiteUrls> {
+    // Tries to get the site urls in the following order:
+    // 1. From the deployments database table
+    // 2. From the config.yml file
+    // 3. From the GitHub repository description
+    // Otherwise, returns an empty string for the url
+    const { siteName } = sessionData
+    let configYmlData: ConfigYmlData = {}
+    let repoDescTokens: string[] = []
+
+    const site = await this.siteRepository.findOne({
+      where: { name: siteName },
+      include: {
+        model: Deployment,
+        as: "deployment",
+      },
+    })
```

Comment:
Actually, given that this is the SitesService itself, does this suggestion mean that we would need to query SitesService for the existence of the site, before we can call this endpoint? Also, this might cause duplicated code if we want to reuse this method (e.g. if multiple methods call `getUrlsOfSite`, both would need to first query some generic `SitesService.get` method to retrieve a `Site`, before this method could be called) - I'm not sure if this behaviour is desirable

Reply:
Discussed offline. This is a refactor that we will do separately soon<sup>TM</sup>